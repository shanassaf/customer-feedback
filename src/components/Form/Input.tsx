import React from 'react';
import styled from 'styled-components';

interface Props {
	setFeedback: (feedback: Feedback) => void;
	feedback: Feedback;
	name: string;
	pattern?: string;
	type: string;
}

const StyledInput = styled.input`
	padding: 10px 4px;
	cursor: pointer;

	:hover {
		border-color: #1b047c;
	}
`;

export const Input = ({ setFeedback, feedback, name, pattern, type }: Props): JSX.Element => {
	return (
		<StyledInput
			type={type}
			required
			name={name}
			value={feedback[name]}
			pattern={pattern}
			onChange={(ev) => {
				setFeedback({ ...feedback, [ev.target.name]: ev.target.value });
			}}
		/>
	);
};
