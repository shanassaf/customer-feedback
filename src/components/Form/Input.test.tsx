import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Input } from './Input';

describe('Input', () => {
	const feedback = {
		name: '',
		email: '',
		comment: '',
		rating: '0',
	} as Feedback;
	const props = {
		setFeedback: jest.fn(),
		feedback,
		name: 'name',
		type: 'text',
	};
	test('component renders correctly', () => {
		const { asFragment } = render(<Input {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
	test('component calls setFeedback on change', () => {
		const { getByRole } = render(<Input {...props} />);
		const input = getByRole('textbox');
		fireEvent.change(input, { target: { value: 'checkout' } });
		expect(props.setFeedback).toHaveBeenCalledWith({ ...feedback, name: 'checkout' });
	});
});
