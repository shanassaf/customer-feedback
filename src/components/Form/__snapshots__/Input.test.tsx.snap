// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnylx bzTEwn"
    name="name"
    required=""
    type="text"
    value=""
  />
</DocumentFragment>
`;
